File: src_client/fuhahaEngine/ios/iosPluginPreference.m
#import "./fuhaha-Swift.h"
#include "fuhahaEngine.h"
#include "./native.h"

// ----------------------------------------------------------------
// ----------------------------------------------------------------
// ----------------------------------------------------------------

// 設定読み込み 返値文字列は揮発性バッファで解放禁止
char *platformPluginPreferenceGet(char *key){
	NSString *nsstr1 = [NSString stringWithCString: key encoding: NSUTF8StringEncoding];
	NSString *nsstr2 = [IosPluginPreference platformPluginPreferenceGet: nsstr1];
	if(nsstr2 == NULL){return NULL;}

	char *value = (char*)[nsstr2 UTF8String];
	char *buff = (char*)corePluginUtilMemoryTemporary(strlen(value) + 1);
	strcpy(buff, value);

	return buff;
}

// 設定書き込み
void platformPluginPreferenceSet(char *key, char *value){
	NSString *nsstr1 = [NSString stringWithCString: key encoding: NSUTF8StringEncoding];
	NSString *nsstr2 = [NSString stringWithCString: value encoding: NSUTF8StringEncoding];
	[IosPluginPreference platformPluginPreferenceSet: nsstr1 value: nsstr2];
}

// 設定保存
void platformPluginPreferenceCommit(void){
	[IosPluginPreference platformPluginPreferenceCommit];
}

// ----------------------------------------------------------------
// ----------------------------------------------------------------
// ----------------------------------------------------------------

File: src_client/fuhahaEngine/ios/IosPluginPreference.swift
import Foundation

// ----------------------------------------------------------------
// ----------------------------------------------------------------
// ----------------------------------------------------------------

// プラグインクラス
class IosPluginPreference: NSObject{
	// ----------------------------------------------------------------

	// 設定読み込み
	@objc static internal func platformPluginPreferenceGet(_ key: String) -> NSString?{
		let userDefaults = UserDefaults.standard;
		return userDefaults.object(forKey: key) as! NSString?;
	}

	// 設定書き込み
	@objc static internal func platformPluginPreferenceSet(_ key: String, value: String){
		let userDefaults = UserDefaults.standard;
		userDefaults.set(value, forKey: key);
	}

	// 設定保存
	@objc static internal func platformPluginPreferenceCommit(){
		let userDefaults = UserDefaults.standard;
		userDefaults.synchronize();
	}

	// ----------------------------------------------------------------
}

// ----------------------------------------------------------------
// ----------------------------------------------------------------
// ----------------------------------------------------------------

